GoogleUtilities/Network/GULNetworkURLSession.m

```diff
@@ -387,28 +387,20 @@ - (void)URLSession:(NSURLSession *)session
         dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
 
     dispatch_async(evaluateBackgroundQueue, ^{
-      SecTrustResultType trustEval = kSecTrustResultInvalid;
       BOOL shouldAllow;
-      OSStatus trustError;
+      CFErrorRef errorRef = NULL;
 
       @synchronized([GULNetworkURLSession class]) {
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wdeprecated-declarations"
-        trustError = SecTrustEvaluate(serverTrust, &trustEval);
-#pragma clang diagnostic pop
+        shouldAllow = SecTrustEvaluateWithError(serverTrust, &errorRef);
       }
 
-      if (trustError != errSecSuccess) {
-        [self->_loggerDelegate GULNetwork_logWithLevel:kGULNetworkLogLevelError
-                                           messageCode:kGULNetworkMessageCodeURLSession008
-                                               message:@"Cannot evaluate server trust. Error, host"
-                                              contexts:@[ @(trustError), self->_request.URL ]];
-        shouldAllow = NO;
-      } else {
-        // Having a trust level "unspecified" by the user is the usual result, described at
-        // https://developer.apple.com/library/mac/qa/qa1360
-        shouldAllow =
-            (trustEval == kSecTrustResultUnspecified || trustEval == kSecTrustResultProceed);
```

Comment:
From [Apple docs](https://developer.apple.com/documentation/security/2980705-sectrustevaluatewitherror): _The SecTrustEvaluateWithError(_:_:) method returns a pass or fail indicator and an error describing the reason for any failure._  So, I think it's reasonable to rely on the `shouldAllow` result. Furthermore, `GTMSessionFetcher` relies on the API's return bool value:

https://github.com/google/gtm-session-fetcher/blob/4990f8956605007c2546824d57432f04dc37fe70/Sources/Core/GTMSessionFetcher.m#L2615-L2628

Reply:
SGTM! Nice cleanup.